reana_db/models.py

```diff
@@ -492,3 +493,78 @@ class AuditLog(Base, Timestamp):
     def __repr__(self):
         """Audit log string representation."""
         return "<AuditLog {} {}>".format(self.id_, self.action)
+
+
+class ResourceType(enum.Enum):
+    """Enumeration of resource types."""
+
+    cpu = 0
+    gpu = 1
+    disk = 2
+
+
+class ResourceUnit(enum.Enum):
+    """Enumeration of resource usage units."""
+
+    bytes_ = 0
+    milliseconds = 1
+
+
+class Resource(Base, Timestamp):
+    """Resource table."""
+
+    __tablename__ = "resource"
+    __table_args__ = {"schema": "__reana"}
+
+    id_ = Column(UUIDType, primary_key=True, default=generate_uuid)
+    name = Column(String(1024))
+    type_ = Column(Enum(ResourceType), nullable=False)
+    unit = Column(Enum(ResourceUnit), nullable=False)
+    title = Column(String(1024))
```

Comment:
I was thinking `name` being a short descriptive string e.g. "cpu" (that we could also use hard-coded in the application instead of IDs), then `title` being a long human-readable version of the name e.g. "CPU hours at the Foo cluster", so a new `description` column would be actually yet another to add, if we want one.  (BTW, using this logic, `name` could be much shorter column for efficient DB indexing than 1024 string.)

Alternatively, if we never hardcode `cpu`, `gpu` and friends, but we be always looking them up in the application logic already from the get go, e.g. via reana-commons or live DB queries, then it could be sufficient to have only `name` and `description` indeed.  (and no `title`)